### DISCUSSION TITLE
Tasmota for Shelly Plus 2PM German

### DISCUSSION THREAD

**Discussion by Wilhelm1Daniel** [arendst/Tasmota#18273] [Q&A] [answered]
Hi,

I have a question...
is it possible to get the Binary-File "tasmota32solo1.bin" for the Shelly Plus 2PM in german language? It's the only binary which works with this device.

Regards
Daniel

**Comment by Jason2866** [comment, marked as answer]
Yes, just add this to `platformio_tasmota_cenv.ini`
```
[env:tasmota32solo1-DE]
extends                 = env:tasmota32_base
board                   = esp32_solo1
build_flags             = ${env:tasmota32_base.build_flags}
                          -DFIRMWARE_TASMOTA32
                          -DMY_LANGUAGE=de_DE
                          -DCODE_IMAGE_STR='"solo1"'
                          -DOTA_URL='""'
```

and run `pio run -e tasmota32solo1-DE` to compile the solo1 in German.

**Reply by Noschvie** [reply]
Tested running  Docker Tasmota successfully.
But how to use platformio_override.ini for that, outside of directory Tasmota?

**Reply by barbudor** [reply]
@Noschvie not sure to get your point
platformio_override.ini must be in tasmota root folder
As well as platformio_tasmota_cenv.ini

**Reply by Noschvie** [reply]
The question for me: how can this add-on to platformio_tasmota_cenv.ini be persistent to updates of this file.

**Reply by barbudor** [reply]
There is no update to this file, it's yours
A sample file is provided as part of Tasmota repository which you can use as an example.
But this file, as well as user_config_override.h are yours not part of Tasmota. So they will never be overwritten by an update

**Comment by Wilhelm1Daniel** [comment]
For me the Post from Jason2866 works.... THX
But I had to reflash the shelly to Update it, because I couldn't Update it over the Webinterface via file upload.
Is there an another way to Update it with future versions? I wouldn't open my wallswitches everytime.

Regards
Daniel

**Reply by sfromis** [reply]
It should work, depending on exact details of what you did, and your symptoms. While you need the .factory.bin files for wired flashing, upgrade only works when using the regular .bin instead.

**Reply by Wilhelm1Daniel** [reply]
Ok i will Tell you the whole story what I've done until now...
1. Buy the switch, short Google for pinout and tried to Flash with Tasmotizer... FAILED because of ESP32.
2. Tried to Flash with NodeMCU and tasmota32.bin ... FAILED... the shelly didn't work, no WiFi, no AP
3. Tried to Flash with NodeMCU and the original Firmware as bin file...FAILED... the shelly didn't work
5. Google... tried to flash with flash-download-tool and the original Firmware as bin file...FAILED... the shelly didn't work.
6. Google... used MOS to create an upload file with original firmwarefile...FAILED... the shelly didn't work, there was a ShellyAP but i couldn't connect
7. Tried to replace the original firmwarebin in the MOS project with the tasmota32solo1.bin and upload the 5 project files with  flash-download-tool... SUCCESS... the Shelly works with the newest tasmota, but in english 
8. Created a german 32solo1 with the help of Jason2866 
9. Tried 3 times to upload this file via tasmota Web interface...FAILED...the shelly restarts every time in the old english Version
10. Did point 6 again with my created german 32solo...SUCCESS...my shelly works in german now
11. Created a New german 32solo with a few Changes in the tasmota User file and tried to upload like in point 8...FAILED again.
12. Did point 9 again with the New file... SUCCESS

That was the question.... must I everytime Update the shelly wired if a New tasmota Version is released, or is there a way to Update it via file upload or OTA?

Daniel

Sorry for the bad english

**Reply by sfromis** [reply]
The regular setup for flashing Tasmota is that you **have to** use the .factory.bin file for first wired flashing, once and for all to setup the correct partition layout, preferably via the recommended Tasmota Web installer. You should have stayed away from Google, and instead looked at Tasmota docs, to avoid all that trouble and detours, and not have done that setup with 5 project files, which may be the cause of the later trouble.

After having done first flashing with the factory image, you can use the regular non-factory binary for OTA upgrading, via a web server or via file upload. I'm having no trouble with Shelly Plus 2PM upgrades doing it that way.

**Reply by Wilhelm1Daniel** [reply]
ok, will try to use the WebInstaller, but there I only find the normal 32solo1.... where I can find the solo1.factory ?

**Reply by Wilhelm1Daniel** [reply]
Found the right solution... THX to sfromis

1. Upload the 32solo1-factory.bin in german language, created with platformIO via NodeMCU.... SUCCESS
2. Tried to upload the 32solo1 in english from Tasmota-Relase-Page via Webinterface OTA... SUCCESS
3. Upload my created 32solo1 in german via Webinterface File Upload.... SUCCESS

Now I can use both... OTA and file upload...

Is there a way to implement a 32solo1-DE to Tasmota release page that me and other users can use this for OTA?

**Reply by Noschvie** [reply]
Using the Tasmota Web installer you don't need the *factory.bin file, it's handled by the tool.
Give them a try...

**Reply by Wilhelm1Daniel** [reply]
The shelly works now, how I want it.

THX to all

**Reply by sfromis** [reply]
> where I can find the solo1.factory

Same place as the non-factory image: http://ota.tasmota.com/tasmota32/
But still, as Noschvie also mentioned "preferably via the recommended Tasmota Web installer".

Due to the multitude of architectures and languages, the number of precompiled binaries would grow impractical if providing all combinations.

**Reply by Wilhelm1Daniel** [reply]
> Due to the multitude of architectures and languages, the number precompiled binaries would grow impractical if providing all combinations.

ok thats understandable
